sky/engine/tonic/dart_io.cc

```diff
@@ -10,11 +10,21 @@
 
 namespace blink {
 
-void DartIO::InitForIsolate() {
+void DartIO::InitForIsolate(const char* script_uri) {
+  Dart_Handle io_lib = Dart_LookupLibrary(ToDart("dart:io"));
+
   DART_CHECK_VALID(Dart_SetNativeResolver(
-      Dart_LookupLibrary(ToDart("dart:io")),
+      io_lib,
       dart::bin::IONativeLookup,
       dart::bin::IONativeSymbol));
+
+  if (script_uri) {
+    Dart_Handle platform_type = Dart_GetType(io_lib, ToDart("_Platform"),
+                                             0, nullptr);
+    DART_CHECK_VALID(platform_type);
+    DART_CHECK_VALID(Dart_SetField(
+        platform_type, ToDart("_nativeScript"), ToDart(script_uri)));
+  }
```

Comment:
IMHO, we should do this work in DartRuntimeHooks::Install, which is where we do all this similar stuff for the other built-in libraries.